client/src/main/java/plugin/api/API.java

```diff
@@ -309,4 +313,37 @@ public static void SendMessage(String message) {
     public static void RequestNewSong() {
         SoundPlayer.sendTrackEndPacket();
     }
+    public static JagString getHexFromRGB(int red, int green, int blue) {
+        String hexFromRGB = String.format("%02X%02X%02X", red, green, blue);
+        return JagString.parse("<col=" + hexFromRGB + ">");
+    }
+    public static void calcEntityScreenPos(int i, int j, int l) {
+        if (i < 128 || l < 128 || i > 13056 || l > 13056) {
+            ScriptRunner.spriteDrawX = -1;
+            ScriptRunner.spriteDrawY = -1;
+            return;
+        }
+        int i1 = SceneGraph.getTileHeight(plane, l, i) - j;
+        i -= Camera.renderX;
+        i1 -= Camera.anInt40;
+        l -= Camera.renderZ;
+        int j1 = MathUtils.sin[Camera.cameraPitch];
+        int k1 = MathUtils.cos[Camera.cameraPitch];
+        int l1 = MathUtils.sin[Camera.cameraYaw];
+        int i2 = MathUtils.cos[Camera.cameraYaw];
+        int j2 = l * l1 + i * i2 >> 16;
+        l = l * i2 - i * l1 >> 16;
+        i = j2;
+        j2 = i1 * k1 - l * j1 >> 16;
+        l = i1 * j1 + l * k1 >> 16;
+        i1 = j2;
+        // TODO: Add support for HD
+        if (l >= 50) {
+            ScriptRunner.spriteDrawX = Rasteriser.width / 2 + (i << 9) / l;
```

Comment:
You should not be setting internal client values like this as a side effect. The only time API methods should be modifying internal client values is when the API method specifically is declared to do that. This should be using local variables.